### DISCUSSION TITLE
Websocket not connecting to hosted Supabase project

### DISCUSSION THREAD

**Discussion by jonlambert** [supabase/supabase#4754] [Questions] [answered]
Hi!

Super excited to hear RLS is now supported when using Realtime.

I'm attempting to get setup with my experimental hosted Supabase project, but I'm finding it pretty difficult to decipher what I should be doing to get this working. The examples all refer to self-hosted projects.

I'm trying to use the following on the frontend:

```ts
// Where <ProjectID> is my Supabase project ID, and <JWT> is the access_token for the session.
const client = new RealtimeClient('ws://<ProjectID>.supabase.co/realtime/v1', { params: { apiKey: <JWT> } });
client.connect();
```

Which yields:

```
WebSocket connection to 'ws://<ProjectID>.supabase.co/realtime/v1/websocket?apiKey=<JWT>&vsn=1.0.0' failed:
```

I've tried both with and without passing options/JWT, using `wss` and `ws`, but nothing seems to work. Can anybody point me in the right direction? 

Thanks!

**Comment by GaryAustin1** [comment]
It is for sure wss for hosted.  apiKey is your Supabase project anon access key. 
If those don't work, do you have replication on for a table?  (I don't think that is required to connect, but not sure)

**Reply by jonlambert** [reply]
Ah thanks @GaryAustin1, but unfortunately using `wss` and anon key for the API key doesn't seem to work either. Same result! Replication is also enabled for all tables.

**Reply by GaryAustin1** [reply]
Well what you have on the wire looks correct:
wss://zkhfeidkffhblxdfdas.supabase.co/realtime/v1/websocket?apikey=eyJhbhfhfhffhjfjfInR5cCI6IkpXVCJ9.edfegdafdsfxlIjoiYW5vbiIsImlhdCI6MTYzMTgzMDdfsadfsdfiZXhwIjoxgdfTQ3NDA2NDY3fQ.6dpTfElsU7mfdffaMhQqUy-6sXTadfdFodJqcb4o3dkfjjjiej4KtjM&vsn=1.0.0 (not a real keys)
I  use supabase.js though which sets up the realtime connection as part of its process, so not any further help on directly calling it.

**Comment by GaryAustin1** [comment, marked as answer]
@jonlambert try apikey instead of apiKey.

**Reply by jonlambert** [reply]
Cheers @GaryAustin1 - that's got me connected, but I'm not receiving any events yet. I'm assuming this could be RLS kicking in? I've tried using the project's anon JWT as you advised above, then setting the session token afterwards:

```ts
client.setAuth(<AccessTokenJWT>);

const debugMessage = (payload: any) => console.log('Message received:', payload));

// Then subscribing:
socket.channel('realtime:*').on('*', debugMessage);

// With user_token set (found at https://github.com/supabase/realtime#realtime-rls):
socket.channel('realtime:*', { user_token: <AccessTokenJWT> }).on('*', debugMessage);
```

...but nothing is emitted, and I don't see anything in the WebSocket message tab.

This might have to be a new topic but I'd really appreciate any further advice you might have. Thanks!

**Reply by GaryAustin1** [reply]
Yeah you might want to mark this as solved and start a new one, like I said I don't use realtime.js directly, but use supabase.js for my subscribes.

**Reply by GaryAustin1** [reply]
just a note you do need to add .subscribe() after the .on

**Reply by jonlambert** [reply]
Ahhh great point! Thanks @GaryAustin1. Adding `.subscribe()` sends a bunch of messages and looks a lot more promising. I've switched to using `supabase-js` for the subscriptions to see if that works instead, but I'm still receiving no events. 

```ts
client
  .from<Model>('test_table')
  .on('*', payload => console.log('Event received:', payload))
  .subscribe(console.log)
```

I will keep poking around, and if I still can't get it to work I'll open a new discussion. Thanks so much for all your help.
